lib/broccoli/service-worker-manifest.js

```diff
@@ -1,98 +1,88 @@
-"use strict";
-
-var diffingPlugin = require('./diffing-broccoli-plugin');
-var path = require('path');
-var fs = require('fs');
-var crypto = require('crypto');
-
-var FILE_ENCODING = { encoding: 'utf-8' };
-var MANIFEST_FILE = 'manifest.appcache';
-var FILE_HASH_PREFIX = '# sw.file.hash:';
-
-class DiffingSWManifest {
-  constructor(inputPath, cachePath, options) {
-    this.inputPath = inputPath;
-    this.cachePath = cachePath;
-    this.options = options;
-    this.firstBuild = true;
-  }
-
-  rebuild(diff) {
-    var manifest = {};
-    if (this.firstBuild) {
-      this.firstBuild = false;
-    } else {
-      // Read manifest from disk.
-      manifest = this.readManifestFromCache();
+/* jshint node: true, esnext: true */
+'use strict';
+
+const Plugin  = require('broccoli-caching-writer');
+const path    = require('path');
+const fs      = require('fs');
+const crypto  = require('crypto');
+
+const MANIFEST_FILE     = 'manifest.appcache';
+const FILE_HASH_PREFIX  = '# sw.file.hash:';
+
+module.exports = SWManifestPlugin;
+
+SWManifestPlugin.prototype = Object.create(Plugin.prototype);
```

Comment:
You can use `class` in NodeJS.  I don't know how much this plugin needs to be changed, but this looks a bit extreme.

@alxhub: could you take a look for this file's validity please? We're moving to Broccoli 1.